### PR TITLE
Gives roaches and many swarmer types AP

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
Many roaches spiders and termties now have AP with their attacks making tanking up on armor less affective
By Humonitarian
Super Mobs can now have AP with attacking

### CODE REVIEW

**Comment by lolman360** [comment]
isn't the entire point of the weak swarming enemies that they do get defeated by armor

**Comment by cdb-is-not-good** [comment]
> isn't the entire point of the weak swarming enemies that they do get defeated by armor

All the more reason for us to use my suggestion of instead giving them sharp attacks. 

Also please port over Eris' support for mobs using sharp attacks :p

**Comment by ThePainkiller** [comment]
Why

**Comment by lolman360** [comment]
> > isn't the entire point of the weak swarming enemies that they do get defeated by armor
> 
> All the more reason for us to use my suggestion of instead giving them sharp attacks.
> 
> Also please port over Eris' support for mobs using sharp attacks :p

what suggestion

**Comment by cdb-is-not-good** [comment]
> > > isn't the entire point of the weak swarming enemies that they do get defeated by armor
> > 
> > 
> > All the more reason for us to use my suggestion of instead giving them sharp attacks.
> > Also please port over Eris' support for mobs using sharp attacks :p
> 
> what suggestion

My idea was simply to keep their damage at current values but make their attacks sharp.

**Comment by Trilbyspaceclone** [comment]
> > isn't the entire point of the weak swarming enemies that they do get defeated by armor
> 
> All the more reason for us to use my suggestion of instead giving them sharp attacks.
> 
> Also please port over Eris' support for mobs using sharp attacks :p

Eris mob sharp attacks were not added in that pr.
Its outside the scope of this pr

File: code/modules/mob/living/carbon/superior_animal/giant_spider/types/hunter.dm [hunk 1]
```diff
@@ -13,6 +13,7 @@
 	meat_type = /obj/item/reagent_containers/food/snacks/meat/spider/hunter
 	meat_amount = 4
 	emote_see = list("chitters.","rubs its legs.","bounces in place.")
+	armor_penetration = 10
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/superior_animal/giant_spider/types/hunter.dm:16]:
```suggestion
```

File: code/modules/mob/living/carbon/superior_animal/giant_spider/types/hunter.dm [hunk 2]
```diff
@@ -69,6 +70,7 @@
 	health = 140
 	melee_damage_lower = 20
 	melee_damage_upper = 25
+	armor_penetration = 15
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/superior_animal/giant_spider/types/hunter.dm:73]:
```suggestion
```

File: code/modules/mob/living/carbon/superior_animal/giant_spider/types/hunter.dm [hunk 3]
```diff
@@ -86,6 +88,7 @@
 	meat_type = /obj/item/reagent_containers/food/snacks/meat/spider/plasma
 	meat_amount = 4
 	emote_see = list("chitters.","rubs its legs.","vibrates.")
+	armor_penetration = 50
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/superior_animal/giant_spider/types/hunter.dm:91]:
```suggestion
```

File: code/modules/mob/living/carbon/superior_animal/giant_spider/types/hunter.dm [hunk 4]
```diff
@@ -105,6 +108,7 @@
 	emote_see = list("chitters.","rubs its legs.","thumps its many legs on the ground.")
 	mob_size = MOB_LARGE
 	armor = list(melee = 15, bullet = 15, energy = 0, bomb = 5, bio = 10, rad = 25, agony = 0)
+	armor_penetration = 10
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/superior_animal/giant_spider/types/hunter.dm:111]:
```suggestion
```

File: code/modules/mob/living/carbon/superior_animal/giant_spider/types/hunter.dm [hunk 5]
```diff
@@ -210,13 +214,15 @@
 	move_to_delay = 4
 	melee_damage_lower = 35
 	melee_damage_upper = 40
+	armor_penetration = 20
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/superior_animal/giant_spider/types/hunter.dm:217]:
```suggestion
```

File: code/modules/mob/living/carbon/superior_animal/giant_spider/types/hunter.dm [hunk 6]
```diff
@@ -210,13 +214,15 @@
...
 
 /mob/living/carbon/superior_animal/giant_spider/tarantula/burrowing
 	name = "trapdoor spider"
 	desc = "Furry and brown, it makes you shudder to look at it. Tough, durable, and strong. Unlike the usual strong-bodied spiders, this one carries sleep toxin in its deadly fangs."
 	icon_state = "burrowing"
 	icon_living = "burrowing"
 	poison_type = "stoxin"
+	armor_penetration = 0
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/superior_animal/giant_spider/types/hunter.dm:225]:
```suggestion
```

File: code/modules/mob/living/carbon/superior_animal/giant_spider/types/nurse.dm [hunk 7]
```diff
@@ -25,6 +25,7 @@
 	var/egg_inject_chance = 0 //AHAHAHAHAHAHAHAAHAHAH, no
 	life_cycles_before_sleep = 3000 //We need more time to eat and web
 	inherent_mutations = list(MUTATION_PROT_MILK, MUTATION_SPIDER_FRIEND, MUTATION_NERVOUSNESS, MUTATION_DEAF)
+	armor_penetration = 30
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/superior_animal/giant_spider/types/nurse.dm:28]:
```suggestion
```

File: code/modules/mob/living/carbon/superior_animal/giant_spider/types/nurse.dm [hunk 8]
```diff
@@ -38,6 +39,7 @@
 	poison_per_bite = 4
 	egg_inject_chance = 5 //Yes
 	meat_type = /obj/item/reagent_containers/food/snacks/meat/spider/midwife
+	armor_penetration = 35
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/superior_animal/giant_spider/types/nurse.dm:42]:
```suggestion
```

File: code/modules/mob/living/carbon/superior_animal/giant_spider/types/nurse.dm [hunk 9]
```diff
@@ -52,6 +54,7 @@
 	poison_type = "frostoil"
 	move_to_delay = 6 // Very slow
 	meat_type = /obj/item/reagent_containers/food/snacks/meat/spider/cave_spider
+	armor_penetration = 5
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/superior_animal/giant_spider/types/nurse.dm:57]:
```suggestion
```

File: code/modules/mob/living/carbon/superior_animal/giant_spider/types/nurse.dm [hunk 10]
```diff
@@ -61,6 +64,7 @@
 	deathmessage = "splits open! Several wriggling spiders crawl from its gore!"
 	egg_inject_chance = 2 //maybe...
 	var/has_made_spiderlings = FALSE
+	armor_penetration = 20
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/superior_animal/giant_spider/types/nurse.dm:67]:
```suggestion
```

File: code/modules/mob/living/carbon/superior_animal/giant_spider/types/nurse.dm [hunk 11]
```diff
@@ -89,6 +93,7 @@
 	emote_see = list("chitters.","rubs its legs.","trails webs through its hairs.","screeches.","bounces happily in place!")
 	web_activity = 90
 	armor = list(melee = 15, bullet = 10, energy = 0, bomb = 5, bio = 10, rad = 25, agony = 0)
+	armor_penetration = 40
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/superior_animal/giant_spider/types/nurse.dm:96]:
```suggestion
```

File: code/modules/mob/living/carbon/superior_animal/giant_spider/types/nurse.dm [hunk 12]
```diff
@@ -105,6 +110,7 @@
 	egg_inject_chance = 15 //Defiently
 	//Giving the recluse its own special meat that has zombie powder. Reducing the amount of meat made since this is some hard stuff and the recluse is easy to kill.
 	poison_type = "zombiepowder"
+	armor_penetration = 70
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/superior_animal/giant_spider/types/nurse.dm:113]: Actually okay with this one, these are meant to be scary and one hit is already an instant down.

**Review by cdb-is-not-good — COMMENTED** [review]
Generally not a big fan of giving spiders AP seeing as how we just added in a change that makes them do less vemon if you have more armor - Also they already have quite an edge(variety of effects, explosions, poison, more damage/health) , whereas roaches do not.

**Comment by Trilbyspaceclone** [comment]
Spiders AP dosnt do anything with their injection of toxins code.
Having 40 melee and then 70 AP still makes the armor block the toxins

**Comment by cdb-is-not-good** [comment]
> Spiders AP dosnt do anything with their injection of toxins code. Having 40 melee and then 70 AP still makes the armor block the toxins

Ehhh.....fair enough, i'd SEVERELY reduce their AP per this PR, given the buffs that spiders already received. I feel like this would SIGNIFICANTLY strengthen spiders.

File: code/modules/mob/living/carbon/superior_animal/roach/types/kaiser.dm [hunk 13]
```diff
@@ -29,6 +29,7 @@ Has ability of every roach.
 	mouse_opacity = MOUSE_OPACITY_OPAQUE // Easier to click on in melee, they're giant targets anyway
 
 	flash_resistances = 9.9 // were not fully flash proof but almost...
+	armor_penetration = 75
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/superior_animal/roach/types/kaiser.dm:32]:
```suggestion
	armor_penetration = 40
```
We don't want them to completely invalidate armor

File: code/modules/mob/living/carbon/superior_animal/roach/types/nanites.dm [hunk 14]
```diff
@@ -22,6 +22,7 @@
 
 	min_air_pressure = 0
 	min_bodytemperature = 0
+	armor_penetration = 50
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/superior_animal/roach/types/nanites.dm:25]:
```suggestion
	armor_penetration = 30
```

File: code/modules/mob/living/carbon/superior_animal/roach/types/toxic.dm [hunk 15]
```diff
@@ -13,6 +13,7 @@
 	knockdown_odds = 3
 	melee_damage_lower = 3
 	melee_damage_upper = 7 //Weaker than hunter
+	armor_penetration = 35
```
Comment (cdb-is-not-good) [inline code/modules/mob/living/carbon/superior_animal/roach/types/toxic.dm:16]:
```suggestion
	armor_penetration =25
```

**Review by cdb-is-not-good — APPROVED** [review]

**Pull request merged**